code/code/misc/combat.cc

```diff
@@ -3902,22 +3902,22 @@ int TBeing::oneHit(TBeing *vict, primaryTypeT isprimary, TThing *weapon, int mod
       if (monkDodge(vict, weapon, &dam, w_type, part_hit))
         mess_sent |= ONEHIT_MESS_DODGE;
       else if (thiefDodge(vict, weapon, &dam, w_type, part_hit))
-	mess_sent |= ONEHIT_MESS_DODGE;
+	      mess_sent |= ONEHIT_MESS_DODGE;
       else if (parryWarrior(vict, weapon, &dam, w_type, part_hit)){
-	if(doesKnowSkill(SKILL_RIPOSTE) &&   // must know the skill
-	 (::number(0,99) < 50) &&                // only 50% of the time
-	 bSuccess(SKILL_RIPOSTE)){
-	  act("$n uses $s parry to execute a riposte.",
-	      FALSE, this, 0, vict, TO_CHAR, ANSI_PURPLE);
-	  act("You use your parry to execute a riposte.", 
-	      FALSE, this, 0, vict, TO_VICT, ANSI_PURPLE);
-	  act("$n uses $s parry to execute a riposte.",
-	      FALSE, this, 0, vict, TO_NOTVICT);
-	  // actual ripose attack is added in hit().  Klugey, but
-	  // easier than trying to create a seperate attack here.
-	  SET_BIT(vict->specials.affectedBy, AFF_RIPOSTE);
-	}
-	mess_sent |= ONEHIT_MESS_DODGE;
+	      if(vict->doesKnowSkill(SKILL_RIPOSTE) &&   // must know the skill
+	        (::number(0,99) < 50) &&                // only 50% of the time
+	        vict->bSuccess(SKILL_RIPOSTE)){
```

Comment:
I agree that the victim's riposte skill should be checked, not the attacker's — however, how does that change the actor in message?

Reply:
woops, no change to message.  ammended commit

Reply:
Can't see the fix to message, again